### PR TITLE
Nodify, bundle and deprecate old files

### PR DESCRIPTION
**Pull request by alshakero**
Related to https://github.com/Palindrom/JSON-Patch-OT/issues/5.

This is 100% backward compatible. Yet 100% Node and future Palindrom compatible.

Tested with Browser `<script />`, Node CommonJS, Babel ES6 imports and TS imports. All Jasmine tests pass on Node and browser.

With great Typings files, thanks to @tomalec awesome JSDoc.

Usage examples (tested each):
```html
<script src='dist/json-patch-queue-synchronous.min.js'></script>
```
or
```html
<script src='dist/json-patch-queue.min.js'></script>
```

in Node's CommonJS
```js
var JSONPatchQueue = require('json-patch-queue').JSONPatchQueue;
```
Or
```js
var JSONPatchQueue = require('json-patch-queue').JSONPatchQueueSynchronous;
```
in Node's TS and ES6
```js
import { JSONPatchQueue } from 'json-patch-queue';
```
or
```js
import { JSONPatchQueueSynchronous } from 'json-patch-queue';
```

### CODE REVIEW

**Comment by alshakero** [comment]
@tomalec ready for review.

File: gruntfile.js [hunk 1]
```diff
@@ -1,45 +0,0 @@
-module.exports = function(grunt) {
```
Comment (tomalec) [inline gruntfile.js]: Don't we need bump task any longer?

**Review by tomalec — CHANGES_REQUESTED** [review]
For me it looks kinde hacky that just ot be able to do https://github.com/Palindrom/JSON-Patch-Queue/pull/4/files#diff-eb50505eb60eb228148efeebce771b5f

Write 
```js
module.exports = JSONPatchQueue;
module.exports.default = JSONPatchQueue; 
```
instead of 
```js
if (typeof module !== "undefined") {
    module.exports = JSONPatchQueue;
}
```

We need to load 75lines = 28% code more in the browser.
https://github.com/Palindrom/JSON-Patch-Queue/pull/4/files#diff-8abfd87d68e18e03b76cb3f3591fc129

**Comment by tomalec** [comment]
I'd rather expect it to be:

1. Plain source file without any webpack related stuff, that could be easily maintained and loaded as good old, non-bundled `<script src='src/json-patch-queue.js'></script>`
2. module ready version that works with node, commonjs, babel, ts, liek  `<script src='dist/json-patch-queue.js'></script>`, as this is the version which requires just 4 lines more, comparing to source
3. Bundled version with all deps (not applicable here, as this one has none) ready for browsers

File: gruntfile.js (hunk 1 above)
Comment (alshakero) [reply gruntfile.js]: Well, it's two files that we need to bump. And I thought Grunt with Webpack is too much. They share a lot of the functionality.

**Comment by alshakero** [comment]
@tomalec done. I tested again all the above tests and copied HTML spec runner to test minified files.

File: bower.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "json-patch-queue",
-  "version": "1.0.0",
+  "version": "2.0.0",
```
Comment (tomalec) [inline bower.json]: have you already bumped the version

**Review by tomalec — CHANGES_REQUESTED** [review]
I would bump the version after merge, with single bump commit.
Other than that LGTM.

**Review by tomalec — APPROVED** [review]

**Comment by tomalec** [comment]
LGTM, please merge, and bump major version

**Pull request merged**